packages/node_modules/@node-red/editor-client/src/js/ui/editor.js

```diff
@@ -136,22 +140,64 @@ RED.editor = (function() {
         if (/^\$\{[a-zA-Z_][a-zA-Z0-9_]*\}$/.test(value)) {
             return true;
         }
+        var label = null;
+        if (("label" in definition[property]) &&
+            ((typeof definition[property].label) == "string")) {
+            label = definition[property].label;
+        }
         if ("required" in definition[property] && definition[property].required) {
             valid = value !== "";
+            if (!valid && label) {
+                return RED._("validator.errors.missing-required-prop", {
+                    prop: label
+                });
+            }
         }
         if (valid && "validate" in definition[property]) {
             try {
-                valid = definition[property].validate.call(node,value);
+                var opt = {};
+                valid = definition[property].validate.call(node,value, opt);
+                if ((typeof valid) === "string") {
+                    return valid;
+                }
```

Comment:
It is possible that an existing validate function returns a String and relies on `""` being false and anything else being true.

This code would then cause it to incorrectly mark the property as invalid. We discussed in the forum using the new second argument as a sign the validate function returns strings to mean invalid.

```suggestion
                valid = definition[property].validate.call(node,value, opt);
                // If the validator takes two arguments, it is a 3.x validator that
                // can return a String to mean 'invalid' and provide a reason
                if (definition[property].validate.length === 2 && (typeof valid) === "string") {
                    return valid;
                }
```

Reply:
Sorry for late reply.
I have merged your suggested code. 
Regarding the built-in validator, I have changed it to use the label property of property definition.